irr/include/CMeshBuffer.h

```diff
@@ -170,16 +170,21 @@ class CMeshBuffer : public IMeshBuffer
 			return;
 
 		const u32 vertexCount = getVertexCount();
-		u32 i;
 
-		Vertices.reallocate(vertexCount + numVertices);
-		for (i = 0; i < numVertices; ++i) {
+		// Only reallocate if there are enough new vertices. Otherwise append()
```

Comment:
this implementation is bad anyway
with a `std::vector` I'd do this:
```cpp
Vertices.insert(Vertices.end(), static_cast<const T *>(vertices), static_cast<const T *>(vertices) + numVertices);
```
and it would automatically know how much to extend the data buffer

Reply:
We could merge this fix for backporting purposes but I plan to get rid of `core::array` in this class anyway.

Reply:
I thought about using ::insert, but for the indices, we don't insert the values themselves, but add the old topmost vertexindex to them, which would make it awkward.

One could also just push, and not explicitly reallocate. But the exact thing done here doesn't matter anyways, and I don't want to recompile half of minetest again. :shrug: 
(Feel free to push to the branch if you want to.)

(I haven't replaced the irr array with a std::vector, because I wanted to avoid touching all the things. (`Vertices` is a public member.))

Reply:
I'll open a separate PR.